frontend/src/metabase/lib/i18n.js

```diff
@@ -190,3 +193,17 @@ if (window.MetabaseSiteLocalization) {
 if (window.MetabaseUserLocalization) {
   setLocalization(window.MetabaseUserLocalization);
 }
+
+/**
+ * In static embeddings/public links, there is no user locale, since there is no user in static embeddings/public links.
+ * But we reset the locale to the site locale when `withInstanceLanguage` is called. This breaks static embeddings/public links
+ * since they don't have a user locale. So this function is for them to set the locale from a URL hash as the user locale,
+ * then the translation on some part of FE still works even after `withInstanceLanguage` is called.
+ *
+ * @param {object} translationsObject A translated object with the same structure as the one produced in `loadLocalization` function.
+ */
+export function setUserLocale(translationsObject) {
+  if (!isEmbeddingSdk) {
```

Comment:
This is new. We shouldn't pollute SDK users' `window` object.

I'll open the same change to master as well.

Reply:
Opened https://github.com/metabase/metabase/pull/49770